lib/bolt/apply_target.rb

```diff
@@ -48,7 +50,7 @@ def initialize(target_hash, config)
       # Merge the config hash with inventory config
       config = Bolt::Util.deep_merge(config, @config || {})
```

Comment:
This made me thing that empty hash should be valide. https://github.com/puppetlabs/bolt/blob/62e3e66468502763fc699b6170393f984d8f9ce3/lib/bolt/apply_inventory.rb#L18 Where does `@config` come from?

Reply:
It comes from the `instance_variable_set` loop above.

Reply:
Sorry yeah that was a vague question. If i understand correctly `@config` is something set on a Target's config section (either in a plan with a target method or on a target's config section in inventory) and the `config_hash` is "global" config set on inventory.yaml/bolt.yaml (ie *not*) set on an individual Target object.